src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/ActivityMappingService.kt

```diff
@@ -82,12 +82,64 @@ class ActivityMappingService(
     }
   }
 
+  suspend fun updateScheduleMappings(updateRequest: ActivityMappingDto): ActivityMappingDto {
+    val activityScheduleId = if (activityMappingRepository.existsById(updateRequest.activityScheduleId)) {
```

Comment:
this reads slightly oddly as you're not using the result of the repository call to set the variable.  Might be better to have an `! existsById` `if` check and throw and then following line sets the `val`